decentraland/screens/ListLandForSaleScreen.js

```diff
@@ -20,17 +21,121 @@ export class ListLandForSaleScreen extends React.Component {
     } = this.props;
     try {
       showInfo("Loading land for sale...");
-      const assetsForSale = await getAllAssetsForSale();
-      for (let promise of assetsForSale) {
-        const assetForSale = await promise;
-        appendLandForSaleToList(assetForSale);
-      }
+      const assetsForSale = await this._getAllAssetsForSale();
+
+      // Non-functional style
+      // const loadingAssetsOnScreen = [];
+      // for (let promise of assetsForSale) {
+      //   let loadAssetToScreen = async () => {
+      //     const assetForSale = await promise;
+      //     appendLandForSaleToList(assetForSale);
+      //   };
+      //   loadingAssetsOnScreen.push(loadAssetToScreen());
+      // }
+
+      // Functional style
+      // const loadingAssetsOnScreen = assetsForSale.map(promise => {
+      //   let loadAssetOnScreen = async () => {
+      //     const assetForSale = await promise;
+      //     appendLandForSaleToList(assetForSale);
+      //   };
+      //   return loadAssetOnScreen();
+      // });
+
+      // Functional style with unnamed function
+      // const loadingAssetsOnScreen = assetsForSale.map(promise =>
+      //   (async () => {
+      //     const assetForSale = await promise;
+      //     return appendLandForSaleToList(assetForSale);
+      //   })()
+      // );
+
+      // Functional style with unnamed function - short
+      const loadingAssetsOnScreen = assetsForSale.map(promise =>
+        (async () => appendLandForSaleToList(await promise))()
+      );
+
+      await Promise.all([...loadingAssetsOnScreen]);
       setLoadingAssetsForSaleInProgress(false);
     } catch (err) {
       showError(err);
     }
   };
 
+  // Note: Returns a list of Promises
+  _getAllAssetsForSale = async () => {
+    const decentralandUtils = new DecentralandUtils();
+    const { getAllAssetsForSale: getAllOpenSellOrders } = decentralandUtils;
+
+    const openSellOrders = await getAllOpenSellOrders();
+
+    const contracts = getContracts();
+    const { estateContract } = contracts;
+
+    const estatesForSale = [];
+    for (let order of openSellOrders) {
+      const { nftAddress } = order;
+      const isEstate = addressesAreEqual(
+        nftAddress,
+        estateContract.getAddress()
+      );
+      if (isEstate) estatesForSale.push(order);
+    }
+
+    const assetsForSale = [];
+    // Note: Getting only the first 10 assets for now
+    // See more: https://github.com/tasitlabs/tasit/issues/155
+    const listSize = 10;
+    for (let order of estatesForSale.slice(0, listSize)) {
+      let assetForSalePromise = this._toAssetForSale(order);
+      assetsForSale.push(assetForSalePromise);
+    }
+
+    return assetsForSale;
+  };
+
+  _toAssetForSale = async sellOrder => {
```

Comment:
I almost wonder if `reformatAsset` might be a clearer name. When I see the to it makes me wonder what the from is, and I assume from the current name it doesn't involve and network I/O, but a name like reformat makes that very clear

Reply:
I agree (that this name seems that it involves an I/O operation).
My intention here was something like `fromSellOrderToAssetForSale` because of I has assumed the name `sellOrder` to "raw data" from blockchain and `assetForSale` for the object used by the app as a way to avoid naming confusion.
Maybe the methods names from `DecentralandUtils` could be changed since its methods are returning "raw data".
Note: I'm used `raw data` to refs to data getted by the contract events, often `raw` data is related to hex data.

Reply:
That all makes sense. Sometimes I even just use a `raw` prefix in the name but have the rest of the name be the same. So like `rawAssetForSale`

Reply:
This change won't gate merging

Reply:
Extracted to: https://github.com/tasitlabs/tasit/issues/273

Reply:
Thanks! Resolving